bobtask/map.go

```diff
@@ -96,3 +97,29 @@ func (tm Map) KeysSortedAlpabethically() (keys []string) {
 	sort.Strings(keys)
 	return keys
 }
+
+// CollectTasksInPipeline will collect all task names in the pipeline for task taskName
+// in the tasksInPipeline slice
+func (tm Map) CollectTasksInPipeline(taskName string, tasksInPipeline *[]string) error {
```

Comment:
I think it's no necessary to pass a slice explicitly by pointer `*[]string` as a slice is always passed by pointer (correct me if i'm wrong). The only valid reason to use a pointer to a slice if you want to be it a optional parameter for serialization/marshaling.

Reply:
Actually in this case is nedeed. Slice is passed by pointer, but when using append if cap == len then a new internal array is allocated so the pointer will change.

More info here: https://medium.com/swlh/golang-tips-why-pointers-to-slices-are-useful-and-how-ignoring-them-can-lead-to-tricky-bugs-cac90f72e77b